Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -0,0 +1,217 @@
+        /// Called by GetName.
+        /// </summary>
+        /// <returns>
+        /// Returns the first of these that is not null or empty:
+        /// - Value returned by the base implementation
+        /// - Name of the owning CarouselItem
+        /// - Carousel class name
+        /// </returns>
+        protected override string GetNameCore()
+        {
+            string name = AutomationProperties.GetName(this.OwnerCarouselItem);
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            name = this.OwnerCarouselItem.Name;
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            var textBlock = this.OwnerCarouselItem.FindDescendant<TextBlock>();
+            if (textBlock != null)
+            {
+                return textBlock.Text;
+            }
+
+            name = base.GetNameCore();
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Called by GetAutomationId that gets the **AutomationId** of the element that is associated with the automation peer.
+        /// </summary>
+        /// <returns>
+        /// The string that contains the automation ID.
+        /// </returns>
+        protected override string GetAutomationIdCore()
+        {
+            var automationId = base.GetAutomationIdCore();
+            if (!string.IsNullOrEmpty(automationId))
+            {
+                return automationId;
+            }
+
+            if (this.OwnerCarouselItem != null)
+            {
+                return this.GetNameCore();
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Gets the control pattern that is associated with the specified Windows.UI.Xaml.Automation.Peers.PatternInterface.
+        /// </summary>
+        /// <param name="patternInterface">A value from the Windows.UI.Xaml.Automation.Peers.PatternInterface enumeration.</param>
+        /// <returns>The object that supports the specified pattern, or null if unsupported.</returns>
+        protected override object GetPatternCore(PatternInterface patternInterface)
+        {
+            switch (patternInterface)
+            {
+                case PatternInterface.SelectionItem:
+                    return this;
+            }
+
+            return base.GetPatternCore(patternInterface);
+        }
+
+        /// <summary>
+        /// Returns the size of the set where the element that is associated with the automation peer is located.
+        /// </summary>
+        /// <returns>
+        /// The size of the set.
+        /// </returns>
+        protected override int GetSizeOfSetCore()
+        {
+            int sizeOfSet = base.GetSizeOfSetCore();
+
+            if (sizeOfSet != -1)
+            {
+                return sizeOfSet;
+            }
```

Comment:
Was there a specific use case you had in mind for that code? I think that this code only makes a difference if AutomationProperties.SizeOfSet was set which the Canvas control does not do.

Reply:
This was to ensure that the value is set for the size of the set. When testing this locally without, the values were not returned, assumably because the value is always -1. That's also what this check is for, if the size of the set has already been defined, it will return that value instead.

Reply:
But who would set the PositionInSet/SizeOfSet property beside the canvas?

Reply:
I think this is an assumption though. I do agree that it's highly unlikely to happen, but it can, and in those circumstances, it would be best that the developer has the value they set returned than the control doing something unexpected.